Source/Csla/Configuration/Fluent/SerializationOptions.cs

```diff
@@ -7,44 +7,41 @@
 //-----------------------------------------------------------------------
 
 using Csla.Serialization;
-using Csla.Serialization.Mobile;
+using Microsoft.Extensions.DependencyInjection.Extensions;
 
 namespace Csla.Configuration
 {
   /// <summary>
   /// Use this type to configure the settings for serialization.
   /// </summary>
-  public class SerializationOptions
+  /// <param name="cslaOptions">The CSLA options.</param>
+  public class SerializationOptions(CslaOptions cslaOptions)
   {
+    /// <summary>
+    /// Gets the CSLA .NET configuration options.
+    /// </summary>
+    protected CslaOptions CslaOptions => cslaOptions;
+
     /// <summary>
     /// Sets the serialization formatter type used by CSLA .NET
     /// for all explicit object serialization (such as cloning,
     /// n-level undo, etc). Default is MobileFormatter.
     /// </summary>
-    public SerializationOptions SerializationFormatter<T>() where T: ISerializationFormatter
+    public SerializationOptions UseSerializationFormatter<T>() where T : ISerializationFormatter
     {
-      ApplicationContext.SerializationFormatter = typeof(T);
+      CslaOptions.Services.TryAddTransient(typeof(ISerializationFormatter), typeof(T));
```

Comment:
This should just add the formatter specified. `CslaOptions.Services.AddTransient<ISerializationFormatter, T>()`. It's up to the user to use the configuration correctly. It would be very awkward why my call did not register the type.